rust/worker/src/server.rs

```diff
@@ -136,562 +135,241 @@ impl WorkerServer {
-            }
-        };
-
-        let mut proto_results_for_all = Vec::new();
-
-        let mut query_vectors = Vec::new();
-        for proto_query_vector in request.vectors {
-            let (query_vector, _encoding) = match proto_query_vector.try_into() {
-                Ok((vector, encoding)) => (vector, encoding),
-                Err(e) => {
-                    return Err(Status::internal(format!("Error converting vector: {}", e)));
-                }
-            };
-            query_vectors.push(query_vector);
-        }
-
-        let dispatcher = match self.dispatcher {
-            Some(ref dispatcher) => dispatcher.clone(),
-            None => {
-                return Err(Status::internal("No dispatcher found"));
-            }
-        };
-
-        let system = match self.system {
-            Some(ref system) => system,
-            None => {
-                tracing::error!("No system found");
-                return Err(Status::internal("No system found"));
-            }
-        };
-
-        let knn_filter_orchestrator = KnnFilterOrchestrator::new(
-            self.blockfile_provider.clone(),
-            dispatcher.clone(),
-            // TODO: Load the configuration for this
-            1000,
+        scan: chroma_proto::ScanOperator,
+    ) -> Result<(FetchLogOperator, FetchSegmentOperator), Status> {
+        let collection = scan
+            .collection
+            .ok_or(Status::invalid_argument("Invalid Collection"))?;
+        let collection_uuid = Uuid::parse_str(&collection.id)
+            .map_err(|err| Status::invalid_argument(err.to_string()))?;
+        let knn_uuid = Uuid::parse_str(&scan.knn_id)
+            .map_err(|err| Status::invalid_argument(err.to_string()))?;
+        let metadata_uuid = Uuid::parse_str(&scan.metadata_id)
+            .map_err(|err| Status::invalid_argument(err.to_string()))?;
+        let record_uuid = Uuid::parse_str(&scan.record_id)
+            .map_err(|err| Status::invalid_argument(err.to_string()))?;
+        Ok((
             FetchLogOperator {
                 log_client: self.log.clone(),
+                // TODO: Make this configurable
                 batch_size: 100,
-                start_log_offset_id: log_position as u32 + 1,
+                start_log_offset_id: collection.log_position as u32 + 1,
                 maximum_fetch_count: None,
-                collection_uuid,
+                collection_uuid: CollectionUuid(collection_uuid),
             },
             FetchSegmentOperator {
                 sysdb: self.sysdb.clone(),
-                collection_uuid,
-                collection_version,
-                metadata_uuid: None,
-                record_uuid: None,
-                vector_uuid: Some(IndexUuid(segment_uuid)),
+                collection_uuid: CollectionUuid(collection_uuid),
+                collection_version: collection.version as u32,
+                metadata_uuid: IndexUuid(metadata_uuid),
```

Comment:
IndexUuid feels like its being abused here cc @rescrv

Reply:
I'd much prefer an untyped `Uuid` to a typed `Uuid` that gets misused.  We talked offline and i think the solution here is a different typed UUID for each type.  That may be infeasible and require some care at the boundaries of our system (where we type the UUID).

Reply:
https://github.com/chroma-core/chroma/pull/3099